extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/QuarkusAsyncHealthCheckFactory.java

```diff
@@ -27,7 +31,16 @@ public QuarkusAsyncHealthCheckFactory(Vertx vertx) {
     @Override
     public Uni<HealthCheckResponse> callSync(HealthCheck healthCheck) {
         Uni<HealthCheckResponse> healthCheckResponseUni = super.callSync(healthCheck);
-        return healthCheckResponseUni.runSubscriptionOn(MutinyHelper.blockingExecutor(vertx, false));
+        return healthCheckResponseUni.runSubscriptionOn(command -> {
```

Comment:
We still have a lambda here AFAICS. Don't we want it gone too @cescoffier ?

Also I would advise to use static inner classes to avoid keeping references to the parent class. I don't think it's going to be a problem here as it's a singleton but it hit us quite a few times so I would recommend to be careful about it.

Reply:
+1

Reply:
Hopefully, sync checks will not be more performant than async ones since MutinyHelper is also using lamdas :)